designer/server/src/lib/forms.test.js

```diff
@@ -0,0 +1,521 @@
+import config from '~/src/config.js'
+import { createServer } from '~/src/createServer.js'
+import * as fetch from '~/src/lib/fetch.js'
+import * as forms from '~/src/lib/forms.js'
+import { auth } from '~/test/fixtures/auth.js'
+import { renderResponse } from '~/test/helpers/component-helpers.js'
+
+describe('Forms library routes', () => {
+  /** @type {Server} */
+  let server
+
+  beforeAll(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterAll(async () => {
+    await server.stop()
+  })
+
+  const now = new Date()
+  const authorId = 'f50ceeed-b7a4-47cf-a498-094efc99f8bc'
+  const authorDisplayName = 'Enrique Chase'
+
+  /**
+   * @satisfies {FormMetadataAuthor}
+   */
+  const author = {
+    id: authorId,
+    displayName: authorDisplayName
+  }
+
+  /**
+   * @satisfies {FormMetadata}
+   */
+  const formMetadata = {
+    id: '661e4ca5039739ef2902b214',
+    slug: 'my-form-slug',
+    title: 'Test form',
+    organisation: 'Defra',
+    teamName: 'Defra Forms',
+    teamEmail: 'defraforms@defra.gov.uk',
+    draft: {
+      createdAt: now,
+      createdBy: author,
+      updatedAt: now,
+      updatedBy: author
+    },
+    createdAt: now,
+    createdBy: author,
+    updatedAt: now,
+    updatedBy: author
+  }
+
+  /**
+   * @satisfies {FormDefinition}
+   */
+  const formDefinition = {
+    name: 'Test form',
+    pages: [],
+    conditions: [],
+    sections: [],
+    lists: []
+  }
+
+  describe('Forms library routes (mocked forms functions)', () => {
```

Comment:
Not ideal, but as we were mocking the entire `/forms.js` file, I had to restructure the file with the mocked methods and a separate describe block for the actual methods to test from the file. It probably would've been easier to create a separate sheet in hindsight, but naming might be confusing.